recipes/ncbi-cxx-toolkit-public/26/conanfile.py

```diff
@@ -0,0 +1,1447 @@
+        return self.NCBI_to_Conan_requires.keys()
+
+#----------------------------------------------------------------------------
+    def _translate_ReqKey(self, key):
+        if key in self.NCBI_to_Conan_requires.keys():
+            if key == "BerkeleyDB" and self.settings.os == "Windows":
+                return None
+            if key == "CASSANDRA" and (self.settings.os == "Windows" or self.settings.os == "Macos"):
+                return None
+            if key == "NGHTTP2" and self.settings.os == "Windows":
+                return None
+            return self.NCBI_to_Conan_requires[key]
+        return None
+
+#----------------------------------------------------------------------------
+    @property
+    def _source_subfolder(self):
+        return "src"
+
+#----------------------------------------------------------------------------
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["NCBI_PTBCFG_PACKAGING"] = "TRUE"
+        if self.options.with_projects != "":
+            cmake.definitions["NCBI_PTBCFG_PROJECT_LIST"] = self.options.with_projects
+        if self.options.with_targets != "":
+            cmake.definitions["NCBI_PTBCFG_PROJECT_TARGETS"] = self.options.with_targets
+        return cmake
+
+#----------------------------------------------------------------------------
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 17)
+        if self.settings.os not in ["Linux", "Macos", "Windows"]:   
+            raise ConanInvalidConfiguration("This operating system is not supported")
+        if self.settings.compiler == "Visual Studio" and str(self.settings.compiler.version) < "15":
+            raise ConanInvalidConfiguration("This version of Visual Studio is not supported")
+        if self.settings.compiler == "Visual Studio" and self.options.shared and "MT" in self.settings.compiler.runtime:
+            raise ConanInvalidConfiguration("This configuration is not supported")
+        if self.settings.compiler == "gcc" and str(self.settings.compiler.version) < "7":
+            raise ConanInvalidConfiguration("This version of GCC is not supported")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+#----------------------------------------------------------------------------
+    def requirements(self):
+        NCBIreqs = self._get_RequiresMapKeys()
+        for req in NCBIreqs:
+            pkg = self._translate_ReqKey(req)
+            if pkg is not None:
+                self.requires(pkg)
+
+#----------------------------------------------------------------------------
+    def source(self):
+        tk_url = self.conan_data["sources"][self.version]["url"]
+        print("getting Toolkit sources...")
+        print("from url: " + tk_url)
```

Comment:
Debug output can be removed

```suggestion
```

Reply:
I do not call it "debug output", but, ok, it can be removed